packages/deepl-translation/server/deeplTranslate.js

```diff
@@ -0,0 +1,174 @@
+import {TranslationProviderRegistry, AutoTranslate} from 'meteor/rocketchat:autotranslate';
+import {RocketChat} from 'meteor/rocketchat:lib';
+import _ from 'underscore';
+import s from 'underscore.string';
+
+class DeeplAutoTranslate extends AutoTranslate {
+	/**
+	 * Constructor
+	 */
+	constructor() {
+		super();
+		this.name = 'deepl-translate';
+		this.apiEndPointUrl = 'https://api.deepl.com/v1/translate';
+		// self register & de-register callback - afterSaveMessage based on the activeProvider
+		RocketChat.settings.get('AutoTranslate_ServiceProvider', (key, value) => {
+			if (this.name !== value) {
+				this.deRegisterAfterSaveMsgCallBack(this.name);
+			} else {
+				this.registerAfterSaveMsgCallBack(this.name);
+			}
+		});
+	}
+
+	/**
+	 * Returns metadata information about the service provide
+	 * @protected implements super abstract method.
+	 */
+	_getProviderMetadata() {
+		return {
+			name: this.name,
+			displayName: TAPi18n.__('AutoTranslate_DeepL'),
+			settings: this._getSettings()
+		};
+	}
+
+	/**
+	 * Returns necessary settings information
+	 * about the translation service provider.
+	 * @protected
+	 */
+	_getSettings() {
+		return {
+			apiKey: this.apiKey,
+			apiEndPointUrl: this.apiEndPointUrl
+		};
+	}
+
+	/**
+	 * Return supported languages for translation
+	 */
+	getSupportedLanguages(target) {
+		if (this.autoTranslateEnabled && this.apiKey) {
+			if (!this.supportedLanguages[target]) {
+				return [
+					{
+						'language': 'EN'
+					},
+					{
+						'language': 'DE'
+					},
+					{
+						'language': 'FR'
+					},
+					{
+						'language': 'ES'
+					},
+					{
+						'language': 'IT'
+					},
+					{
+						'language': 'NL'
+					},
+					{
+						'language': 'PL'
+					}
+				];
+			}
+		}
+	}
+
+	translateMessage(message, room, targetLanguage) {
```

Comment:
I'd expect everything above to be part of the super-class. Why is it implemented for each subclass, what's specific about it?

Reply:
Handled.